Expand "Watermark errors" page (#4521)
elastic/docs-content: troubleshoot/elasticsearch/fix-watermark-errors.md

* 75% `none`: The Cloud Deployment's {{es}} node's disk bar turns red, but {{es}} takes no action. {applies_to}`ece: ga` {applies_to}`ess: ga`
* 85% [`low`](elasticsearch://reference/elasticsearch/configuration-reference/cluster-level-shard-allocation-routing-settings.md#cluster-routing-watermark-low): {{es}} stops allocating replica shards and primary shards unless from newly-created indices to the affected node(s).
    ```
    low disk watermark [85%] exceeded on [NODE_ID][NODE_NAME] free: Xgb[X%], replicas will not be assigned to this node
    ```
* 90% [`high`](elasticsearch://reference/elasticsearch/configuration-reference/cluster-level-shard-allocation-routing-settings.md#cluster-routing-watermark-high): {{es}} rebalances shards away from the affected node(s).
    ```
    high disk watermark [90%] exceeded on [NODE_ID][NODE_NAME] free: Xgb[X%], shards will be relocated away from this node
    ```

> * 95% [`flood-stage`](elasticsearch://reference/elasticsearch/configuration-reference/cluster-level-shard-allocation-routing-settings.md#cluster-routing-flood-stage)): {{es}} sets all indices on the affected node(s) to read-only. This is automatically reverted once the affected node’s high disk usage falls below high watermark. 

Reviewer: ```suggestion
* 95% [`flood-stage`](elasticsearch://reference/elasticsearch/configuration-reference/cluster-level-shard-allocation-routing-settings.md#cluster-routing-flood-stage): {{es}} sets all indices on the affected node(s) to read-only. This is automatically reverted once the affected node’s high disk usage falls below high watermark. 
```

    ```
    flood-stage watermark [95%] exceeded on [NODE_ID][NODE_NAME], all indices on this node will be marked read-only
    ```

> 

Reviewer: Suggesting to add the note here. We're calling out this use case while also making the distinction that this not a watermark threshold, but a visual indicator. Please feel free to modify any parts that may be inaccurate.  

```suggestion

:::{note}
:applies_to: { ess:, ece: }

At 75% disk usage, the {{ecloud}} Console displays a red disk indicator for the node to signal elevated usage. This threshold is a visual indicator only and is not tied to any {{es}} watermark or disk-enforcement behavior. No {{es}} allocation or write restrictions are applied at this stage.
:::

```

Does this work?

* From your deployment menu, view the [**Performance**](../../deploy-manage/monitor/access-performance-metrics-on-elastic-cloud.md) page's disk usage chart.
::::::

::::::{applies-item} { self:, eck: }
* (Recommend) Enable [AutoOps](/deploy-manage/monitor/autoops.md).
* Enable [{{es}} monitoring](/deploy-manage/monitor/stack-monitoring.md). When logs and metrics are enabled, monitoring information is visible on {{kib}}'s [Stack Monitoring](/deploy-manage/monitor/monitoring-data/visualizing-monitoring-data.md) page. You can also enable the [Disk usage threshold alert](/deploy-manage/monitor/monitoring-data/configure-stack-monitoring-alerts.md) to be notified about potential issues.
::::::

:::::::

## Monitor rebalancing [fix-watermark-errors-rebalance]

> To verify that shards are moving off the affected node until it falls below high watermark, use the following {{es}} API's:

Reviewer: ```suggestion
To verify that shards are moving off the affected node until it falls below high watermark, use the following {{es}} APIs:
```

:::

## Context

> Elasticsearch uses [disk-based shard allocation watermarks](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-cluster.html#disk-based-shard-allocation) to prevent disk overuse and protect against data loss. Until a node reaches the flood-stage watermark, indexing is not blocked and shards can continue to grow on disk. Default watermark thresholds and their effects:  

Reviewer: We use [absolute paths](https://docs-v3-preview.elastic.dev/elastic/docs-builder/tree/main/syntax/links#absolute-paths) to link to other documents within the same repository. And this is the [cross-repository links](https://docs-v3-preview.elastic.dev/elastic/docs-builder/tree/main/syntax/links#cross-repository-links) syntax.

(Our build tool checks these links to avoid broken links, so it's a better link management strategy). 

In this particular case your link should be:
 
```suggestion
Elasticsearch uses [disk-based shard allocation watermarks](elasticsearch://reference/elasticsearch/configuration-reference/cluster-level-shard-allocation-routing-settings.md) to prevent disk overuse and protect against data loss. Until a node reaches the flood-stage watermark, indexing is not blocked and shards can continue to grow on disk. Default watermark thresholds and their effects:  
```

There are other links in this doc where you should use absolute paths.

```

## Common causes of watermark errors

Watermark errors occur when a node’s disk usage exceeds the configured thresholds (`low`, `high`, or `flood-stage`). While these thresholds protect cluster stability, they can be triggered by several underlying factors including:  

* Sudden ingestion of large volumes of data, often referred to as large indexing bursts, can quickly consume disk space, especially if the cluster is not sized for peak loads. See [Indexing performance considerations](https://www.elastic.co/guide/en/elasticsearch/reference/current/tune-for-indexing-speed.html) for guidance.  

> * Inefficient index settings, unnecessary stored fields, and suboptimal document structures can increase disk consumption. See [Tune for disk usage](https://www.elastic.co/docs/deploy-manage/production-guidance/optimize-performance/disk-usage) for guidance on reducing storage requirements.  

Reviewer: The same with this link.

> * A high number of replicas can quickly multiply storage requirements, as each replica consumes the same disk space as the primary shard. See [Index settings](elasticsearch://reference/elasticsearch/index-settings/index-modules.md) for details.  

Reviewer 2: ```suggestion
* A high number of replicas can quickly multiply storage requirements, as each replica consumes the same disk space as the primary shard. Refer to [Index settings](elasticsearch://reference/elasticsearch/index-settings/index-modules.md) for details.  
```

* Delete indices using the [delete index API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-delete), either permanently if the index isn’t needed, or temporarily to later [restore](../../deploy-manage/tools/snapshot-and-restore/restore-snapshot.md).
* update related [ILM policy](../../manage-data/lifecycle/index-lifecycle-management.md) to push indices through to later [data tiers](../../manage-data/lifecycle/data-tiers.md)

> ## Preventing watermark errors  

Reviewer 2: @rodrigomadalozzo if you feel up for it, we might be able to do two KBs in one go. Supplementing [the one you're doing](https://support.elastic.dev/knowledge/view/6d6a60a5), you could consider (not giving dive-in detail just referencing) the [dangers of bad disk watermarks](https://support.elastic.co/knowledge/76ca0b3b) by adding a bullet point like

```
* {{es}} recommends using default watermark settings. Experts can override [the watermark thresholds and headroom](elasticsearch://reference/elasticsearch/configuration-reference/cluster-level-shard-allocation-routing-settings.md) but risk not giving enough disk for background processes such as force merge, not being right-sized to data ingestion rates vs ILM settings, and possibly `disk is full` errors if 100% disk is reached. 
```

To reduce the likelihood of watermark errors:  

> * Implement more restrictive ILM policies to delete or move data sooner, helping keep disk usage under control. See [Index lifecycle management](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-lifecycle-management.html).  

Reviewer 2: ```suggestion
* Implement more restrictive ILM policies to delete or move data sooner, helping keep disk usage under control. Refer to [Index lifecycle management](/manage-data/lifecycle/index-lifecycle-management.md).
```

* Enable [Autoscaling](/deploy-manage/autoscaling.md) to automatically adjust resources based on storage and performance needs.
* Configure [Stack monitoring](/deploy-manage/monitor/stack-monitoring/ece-ech-stack-monitoring.md) and enable [disk usage monitoring alerts](/solutions/observability/incident-management/alerting.md) to track disk usage trends and identify increases before watermark thresholds are exceeded.

> * Optimize shard sizes to balance disk usage (and performance), avoiding a mix of overly large and small shards. See [Size your shards](/deploy-manage/production-guidance/optimize-performance/size-shards.md).

Reviewer 2: ```suggestion
* Optimize shard sizes to balance disk usage (and performance), avoiding a mix of overly large and small shards. Refer to [Size your shards](/deploy-manage/production-guidance/optimize-performance/size-shards.md).
```
